src/Compilers/CSharp/Portable/Binder/PatternSwitchBinder2.cs

```diff
@@ -0,0 +1,234 @@
+                boundPatternSwitchSectionsBuilder.Add(section);
+            }
+
+            isComplete = defaultLabel != null || subsumption.IsComplete || someCaseMatches;
+            return boundPatternSwitchSectionsBuilder.ToImmutableAndFree();
+        }
+
+        /// <summary>
+        /// Bind the pattern switch section, producing subsumption diagnostics.
+        /// </summary>
+        /// <param name="node"/>
+        /// <param name="originalBinder"/>
+        /// <param name="defaultLabel">If a default label is found in this section, assigned that label</param>
+        /// <param name="someCaseMatches">If a case is found that would always match the input, set to true</param>
+        /// <param name="subsumption">A helper class that uses a decision tree to produce subsumption diagnostics.</param>
+        /// <param name="diagnostics"></param>
+        /// <returns></returns>
+        private BoundPatternSwitchSection BindPatternSwitchSection(
+            SwitchSectionSyntax node,
+            Binder originalBinder,
+            ref BoundPatternSwitchLabel defaultLabel,
+            ref bool someCaseMatches,
+            SubsumptionDiagnosticBuilder subsumption,
+            DiagnosticBag diagnostics)
+        {
+            // Bind match section labels
+            var boundLabelsBuilder = ArrayBuilder<BoundPatternSwitchLabel>.GetInstance();
+            var sectionBinder = originalBinder.GetBinder(node); // this binder can bind pattern variables from the section.
+            Debug.Assert(sectionBinder != null);
+            var labelsByNode = LabelsByNode;
+
+            foreach (var labelSyntax in node.Labels)
+            {
+                LabelSymbol label = labelsByNode[labelSyntax];
+                BoundPatternSwitchLabel boundLabel = BindPatternSwitchSectionLabel(sectionBinder, labelSyntax, label, ref defaultLabel, diagnostics);
+                boundLabelsBuilder.Add(boundLabel);
+            }
+
+            // Bind switch section statements
+            var boundStatementsBuilder = ArrayBuilder<BoundStatement>.GetInstance();
+            foreach (var statement in node.Statements)
+            {
+                boundStatementsBuilder.Add(sectionBinder.BindStatement(statement, diagnostics));
+            }
+
+            return new BoundPatternSwitchSection(node, sectionBinder.GetDeclaredLocalsForScope(node), boundLabelsBuilder.ToImmutableAndFree(), boundStatementsBuilder.ToImmutableAndFree());
+        }
+
+        private BoundPatternSwitchLabel BindPatternSwitchSectionLabel(
+            Binder sectionBinder,
+            SwitchLabelSyntax node,
+            LabelSymbol label,
+            ref BoundPatternSwitchLabel defaultLabel,
+            DiagnosticBag diagnostics)
+        {
+            // Until we've determined whether or not the switch label is reachable, we assume it
+            // is. The caller updates isReachable after determining if the label is subsumed.
```

Comment:
>The caller updates isReachable after determining if the label is subsumed. [](start = 19, length = 74)

Should this be a PROTOTYPE comment? ```isReachable``` is a const, therefore cannot be updated.

Reply:
Update in this case refers to the .Update method of the bound node.

---
In reply to: [156526464](https://github.com/dotnet/roslyn/pull/23209#discussion_r156526464) [](ancestors = 156526464)